sos/plugins/block.py

```diff
@@ -51,17 +51,18 @@ def setup(self):
             "/etc/blkid.tab",
             "/run/blkid/blkid.tab",
             "/proc/partitions",
-            "/proc/diskstats",
-            "/sys/block/*/queue/scheduler"
+            "/proc/diskstats"
         ])
 
         if os.path.isdir("/sys/block"):
             for disk in os.listdir("/sys/block"):
                 if disk.startswith("ram"):
                     continue
                 disk_path = os.path.join('/dev/', disk)
+                queue_path = os.path.join('/sys/block/', disk, 'queue')
                 self.add_udev_info(disk_path)
                 self.add_udev_info(disk_path, attrs=True)
+                self.add_udev_info(queue_path, attrs=True)
```

Comment:
This code is a bit old and crusty: looking at it, I can see where the pattern in the `scsi.py` patch originated. I think a bit of clean up here has been overdue for a while & we can get that done while incorporating these changes. From a quick look I think we can use a similar approach to my suggestion for the other commit (I'm also not really sure we need the `disk_path` udev info twice - `attrs=True` should only add extra information, but perhaps there was a historical reason that people liked the "simpler" data format).

Anyway, either way: the changes are reasonable and we will get them queued up for 3.7 one way or another, but it would be good to address some of the sprawl that has crept into the block plugin over the years as well.

Reply:
Just to chime in here - yes we're grabbing `disk_path` twice because RH support engineers wanted to keep the 'simpler' output as well when we added `add_udev_info()`.

Reply:
That's what I suspected; it makes my kinda sad, but hey ho ;-)

If anything the `-a` output is the more human-readable, as well as including data for the entire device chain: it includes comments, and has whitespace structure designed to make it easy-to-read. In contrast, without `-a` you just get a list of name/value pairs tagged with P/N/S/E etc. indicating the type (physical, name, symlink, env var etc.).

Unfortunately since there isn't any canonical non-file system representation for sysfs data (and since `sysfsutils` is canonically awful), there isn't much we can do.

Reply:
Bryn, Jake, just wanted to update; sorry for the wait.  Upstream, it's not looking good for running udevadm against the queue directory (https://github.com/systemd/systemd/issues/11183).  I've asked Michal for more info in the RH bz (https://bugzilla.redhat.com/show_bug.cgi?id=1657476), but I feel we'll end up having to cp the contents manually.

Reply:
Thanks Bryn, appreciate it. Would guys prefer I rebase/amend, close and send a v2, or just let you guys edit by hand?